src/workflows/airqo_etl_utils/tests/test_fault_detection_utils.py

```
@@ -0,0 +1,169 @@
+    assert faulty_row["battery_low_fault"] == 1
+    assert faulty_row["stuck_value_fault"] == 1
+    assert faulty_row["negative_value_fault"] == 1
+    assert faulty_row["out_of_range_fault"] == 1
+    assert faulty_row["fault_types"] != ""
+
+
+def test_fault_model_can_train_and_score(sample_fault_detection_df, monkeypatch):
+    monkeypatch.setattr("airqo_etl_utils.ml_utils.GCSFileStorage", FakeStorage)
+    monkeypatch.setattr(
+        "airqo_etl_utils.ml_utils.MlflowTracker.log_run",
+        lambda self, **kwargs: None,
```

Ruff (0.15.2)

[warning] Unused lambda argument: `self`

(ARG005)